lib/clAdapters/google/errors.ts

```diff
@@ -0,0 +1,40 @@
+export abstract class GoogleError extends Error {
```

Comment:
could do an initial `instanceof Error` check to see if its a raw error, as the custom "errors" will just be pojos under this method.